x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -86,16 +92,17 @@ export class ReportingPlugin
       const config = reportingCore.getConfig();
 
       const browserDriverFactory = await initializeBrowserDriverFactory(config, logger);
-
-      const esqueue = await createQueueFactory(reportingCore, logger); // starts polling for pending jobs
-      const enqueueJob = enqueueJobFactory(reportingCore, logger); // called from generation routes
+      const store = new ReportingStore(reportingCore, logger);
```

Comment:
How do you feel about adding an instance of the store to the reporting-core object, and having it be access via a getter like we do with the other utilities?

Reply:
++.

We can shrink the options that get passed to `pluginStart` and have the core instance provide those things for itself, and provide getters.

Reply:
For now, my take on this ended up with more refactoring than I intended. Right now, the reporting store is to be set up in a synchronization with:
 - having the config available
 - being made available from ReportingCore before `create_queue` gets called.

A few PRs from now when ESQueue is gone, we can revisit this.